addon/scheduler.ts

```diff
@@ -52,16 +100,30 @@ export function beginTransition(): void {
 }
 
 export function endTransition(): void {
-  _whenRouteDidChange.resolve();
+  if (CAPABILITIES.performanceObserverEnabled) {
+    if (!schedulerOptions.emitMark) {
+      return;
+    }
+
+    _emitMark();
+  } else {
+    _whenRouteDidChange.resolve();
+  }
 }
 
-export function setupRouter(router: Router): void {
+export function setupRouter(router: Router, options: SchedulerOptions): void {
   if (IS_FASTBOOT || (router as any)[APP_SCHEDULER_HAS_SETUP]) {
     return;
   }
 
   (router as any)[APP_SCHEDULER_HAS_SETUP] = true;
 
+  if (CAPABILITIES.performanceObserverEnabled) {
+    schedulerOptions = assign(DEFAULT_OPTIONS, options);
```

Comment:
Should probably not mutate DEFAULT_OPTIONS, right?

```suggestion
    schedulerOptions = assign({}, DEFAULT_OPTIONS, options);
```